cpp/doxygen/Doxyfile

```diff
@@ -817,7 +817,8 @@ WARN_LOGFILE           =
 INPUT                  = main_page.md \
                          regex.md \
                          unicode.md \
-                         ../include
+                         ../include \
+                         ../libcudf_kafka/include
```

Comment:
Did we want this documentation as part of the main libcudf documentation? I don't remember this being discussed.

Reply:
cc @harrism

Reply:
@kkraus14 @harrism had asked me if I could place it there to prevent adding another do them file which I had in my first commit. I’ll let him speak to that however

Reply:
Going to pull this PR and render the docs to see how it looks.

Reply:
After rendering the docs, the only way I could find anything about this was to put "kafka" in the search box. That took me to a `kafka_consumer` class reference page, but the page doesn't appear to be reachable via any of the menus. @jdye64 I think you need to look at our documentation guidelines and add things to groups. See https://github.com/rapidsai/cudf/blob/branch-0.15/cpp/docs/DOCUMENTATION.md

Also, I don't remember the discussion. Why is this in a `namespace external` again?

Reply:
@harrism I will fix the navigation. There really wasn't any discussion on `external`. I had been using that namespace in previous iterations when this was part of the `external_datasources` standalone cmake project I had and the name just sort of carried over. The name doesn't matter to me. I'll call it whatever people want.

Reply:
I created a new 'io_datasources' group next to 'io_readers' and 'io_writers' and added the kafka_consumer to that group.